app/src/main/res/layout/question_player_fragment.xml

```diff
@@ -93,6 +95,38 @@
         app:layout_constraintEnd_toEndOf="parent" />
     </androidx.constraintlayout.widget.ConstraintLayout>
 
+    <FrameLayout
+      android:id="@+id/hints_and_solution_fragment_placeholder"
+      android:layout_width="match_parent"
+      android:layout_height="match_parent">
+    </FrameLayout>
+
+    <FrameLayout
+      android:id="@+id/hints_and_solution_fragment_container"
+      android:layout_width="wrap_content"
+      android:layout_height="wrap_content"
+      android:background="@drawable/hints_background"
+      android:visibility="@{viewModel.isHintBulbVisible() ? View.VISIBLE : View.GONE}"
+      android:layout_gravity="bottom|start">
+
+      <ImageView
+        android:id="@+id/dot_hint"
+        android:layout_width="8dp"
+        android:layout_height="6dp"
+        android:layout_margin="8dp"
+        android:layout_gravity="top|end"
+        android:visibility="@{viewModel.isHintOpenedAndUnRevealed() ? View.VISIBLE : View.GONE}"
+        android:src="@drawable/ic_dot_yellow_24dp"
+        />
```

Comment:
This can be shifted to above line. Ditto elsewhere in this file.

Reply:
Done.